tests/TestCase/Controller/Component/CommonComponentTest.php

```diff
@@ -315,4 +316,45 @@ public function testTrimData() {
 		$this->assertSame($expected, $pass);
 	}
 
+	/**
+	 * @return void
+	 */
+	#[\PHPUnit\Framework\Attributes\DoesNotPerformAssertions]
+	public function testAllowExtensionNone(): void {
+		$this->Controller->Common->allowExtension('');
+	}
+
+	/**
+	 * @return void
+	 */
+	public function testAllowExtensionFail(): void {
+		$this->expectException(NotFoundException::class);
+
+		$this->Controller->Common->allowExtension('csv');
+	}
+
+	/**
+	 * @return void
+	 */
+	#[\PHPUnit\Framework\Attributes\DoesNotPerformAssertions]
+	public function testAllowExtension(): void {
+		$request = $this->Controller->getRequest();
+		$request = $request->withParam('_ext', 'csv');
+		$this->Controller->setRequest($request);
+		$this->Controller->Common->allowExtension('csv');
+	}
+
+	/**
+	 * @return void
+	 */
```

Comment:
I'm not sure what your own standard on this is, but I personally prefer not to need to decipher the test case from a camel-cased method name.

```suggestion
	/**
	 * Test allowing no extension in URL.
     *
	 * @return void
	 */
	#[\PHPUnit\Framework\Attributes\DoesNotPerformAssertions]
	public function testAllowExtensionNone(): void {
		$this->Controller->Common->allowExtension('');
	}

	/**
	 * Test allowing a single extension in URL that was not used
	 *
	 * @return void
	 */
	public function testAllowExtensionFail(): void {
		$this->expectException(NotFoundException::class);

		$this->Controller->Common->allowExtension('csv');
	}

	/***
	 * Test allowing the extension in URL that was used
     *
	 * @return void
	 */
	#[\PHPUnit\Framework\Attributes\DoesNotPerformAssertions]
	public function testAllowExtension(): void {
		$request = $this->Controller->getRequest();
		$request = $request->withParam('_ext', 'csv');
		$this->Controller->setRequest($request);
		$this->Controller->Common->allowExtension('csv');
	}

	/***
	 * Test allowing more than one extension in URL, none was used
     *
	 * @return void
	 */
```